File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if(s.length() != t.length()) return false;
        unordered_map<char,char>mp1;
        unordered_map<char,char>mp2;

        for(int i =0; i<s.length(); i++){
            char a = t[i];
            char b = s[i];
           if(mp1.count(b) && mp1[b] != a){
           return false;
           }
            if(mp2.count(a) && mp2[a] != b){
               return false;
            }
        
            mp1[b] = a;
            mp2[a] = b;
        }
        return true;
        
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length() != t.length()) return false;
        
         int[] cnt1 = new int[256];
         int[] cnt2 = new int[256];

         for(int i=0;i<s.length();i++){
           if(cnt1[s.charAt(i)] != cnt2[t.charAt(i)]) return false;

           cnt1[s.charAt(i)] = i+1;
           cnt2[t.charAt(i)] = i+1;
         }
         

         return true;
        
    }
}